src/garage/tf/policies/categorical_gru_policy2.py

```diff
@@ -0,0 +1,306 @@
+            implies the policy will not be "vectorized", i.e. number of
+            paralle environments for training data sampling = 1.
+
+        Args:
+            dones (numpy.ndarray): Bool that indicates terminal state(s).
+
+        """
+        if dones is None:
+            dones = [True]
+        dones = np.asarray(dones)
+        if self._prev_actions is None or len(dones) != len(self._prev_actions):
+            self._prev_actions = np.zeros(
+                (len(dones), self.action_space.flat_dim))
+            self._prev_hiddens = np.zeros((len(dones), self._hidden_dim))
+
+        self._prev_actions[dones] = 0.
+        self._prev_hiddens[dones] = self.model.networks[
+            'default'].init_hidden.eval()
+
+    def get_action(self, observation):
+        """Return a single action.
+
+        Args:
+            observation (numpy.ndarray): Observations.
+
+        Returns:
+            int: Action given input observation.
+            dict(numpy.ndarray): Distribution parameters.
+
+        """
+        actions, agent_infos = self.get_actions([observation])
+        return actions[0], {k: v[0] for k, v in agent_infos.items()}
+
+    def get_actions(self, observations):
+        """Return multiple actions.
+
+        Args:
+            observations (numpy.ndarray): Observations.
+
+        Returns:
+            list[int]: Actions given input observations.
+            dict(numpy.ndarray): Distribution parameters.
+
+        """
+        if self._state_include_action:
+            assert self._prev_actions is not None
+            all_input = np.concatenate([observations, self._prev_actions],
+                                       axis=-1)
+        else:
+            all_input = observations
+        probs, hidden_vec = self._f_step_prob(all_input, self._prev_hiddens)
+        actions = list(map(self.action_space.weighted_sample, probs))
+        prev_actions = self._prev_actions
+        self._prev_actions = self.action_space.flatten_n(actions)
+        self._prev_hiddens = hidden_vec
+        agent_info = dict(prob=probs)
+        if self._state_include_action:
+            agent_info['prev_action'] = np.copy(prev_actions)
+        return actions, agent_info
+
+    @property
+    def recurrent(self):
+        """Recurrent or not.
+
+        Returns:
+            bool: Whether policy is recurrent or not.
+
+        """
+        return True
+
+    @property
+    def distribution(self):
+        """Policy distribution.
+
+        Returns:
+            tfp.Distribution.OneHotCategorical: Policy distribution.
+
+        """
+        return self.model.networks['default'].dist
+
+    @property
+    def state_info_specs(self):
+        """State info specifcation.
+
+        Returns:
+            List[str]: keys and shapes for the information related to the
+                policy's state when taking an action.
+
+        """
+        if self._state_include_action:
```

Comment:
this isn't covered -- does anything use it?

Reply:
I will add test to cover `self._state_include_action`